src/libaktualizr/storage/fsstorage.cc

```diff
@@ -149,102 +149,98 @@ bool FSStorage::loadTlsCert(std::string* cert) { return loadTlsCommon(cert, conf
 
 bool FSStorage::loadTlsPkey(std::string* pkey) { return loadTlsCommon(pkey, config_.tls_pkey_path); }
 
-void FSStorage::storeMetadata(const Uptane::MetaPack& metadata) {
+void FSStorage::storeMetadataCommon(const Uptane::RawMetaPack& metadata, const std::string& suffix) {
   boost::filesystem::path image_path = Utils::absolutePath(config_.path, config_.uptane_metadata_path) / "repo";
   boost::filesystem::path director_path = Utils::absolutePath(config_.path, config_.uptane_metadata_path) / "director";
 
-  Utils::writeFile((director_path / "root.json"), metadata.director_root.original());
-  Utils::writeFile((director_path / "targets.json"), metadata.director_targets.original());
-  Utils::writeFile((image_path / "root.json"), metadata.image_root.original());
-  Utils::writeFile((image_path / "targets.json"), metadata.image_targets.original());
-  Utils::writeFile((image_path / "timestamp.json"), metadata.image_timestamp.original());
-  Utils::writeFile((image_path / "snapshot.json"), metadata.image_snapshot.original());
+  Utils::writeFile((director_path / ("root.json" + suffix)), metadata.director_root);
+  Utils::writeFile((director_path / ("targets.json" + suffix)), metadata.director_targets);
+  Utils::writeFile((image_path / ("root.json" + suffix)), metadata.image_root);
+  Utils::writeFile((image_path / ("targets.json" + suffix)), metadata.image_targets);
+  Utils::writeFile((image_path / ("timestamp.json" + suffix)), metadata.image_timestamp);
+  Utils::writeFile((image_path / ("snapshot.json" + suffix)), metadata.image_snapshot);
   sync();
 }
 
-bool FSStorage::loadMetadata(Uptane::MetaPack* metadata) {
+bool FSStorage::loadMetadataCommon(Uptane::RawMetaPack* metadata, const std::string& suffix) {
   boost::filesystem::path image_path = Utils::absolutePath(config_.path, config_.uptane_metadata_path) / "repo";
   boost::filesystem::path director_path = Utils::absolutePath(config_.path, config_.uptane_metadata_path) / "director";
 
-  if (!boost::filesystem::exists(director_path / "root.json") ||
-      !boost::filesystem::exists(director_path / "targets.json") ||
-      !boost::filesystem::exists(image_path / "root.json") || !boost::filesystem::exists(image_path / "targets.json") ||
-      !boost::filesystem::exists(image_path / "timestamp.json") ||
-      !boost::filesystem::exists(image_path / "snapshot.json")) {
+  if (!boost::filesystem::exists(director_path / ("root.json" + suffix)) ||
```

Comment:
Why are we aborting if not eveything is present? The only metadata we need to actually have are the two `root.json`s. Everything else can be safely fetched from those alone.

Reply:
So my idea was that it doesn't make sense for a single piece of metadata to be valid or invalid: the verification procedure involves cross-checks between metadata files, so we should treat metadata bundle as an atomic entity. I might be wrong of course, then we need a different logic.

Reply:
Yes, treating metadata packs as atomic is incorrect. See the linked spec below.